### PR TITLE
NITK thesis style

### PR DESCRIPTION
**Pull request by biosainik**
This is a Ph.D. thesis style for NITK Surathkal, India

### CODE REVIEW

**Comment by adam3smith** [comment]
Thanks. I'd recommend taking a closer look at https://github.com/citation-style-language/styles/wiki/Style-Requirements to clarify naming conventions and the content of the info section like id and self link.

**Comment by rmzelle** [comment]
(for starters, the style file name should be something like "national-institute-of-technology-karnataka.csl"; https://github.com/citation-style-language/styles/wiki/Style-Requirements#4---file-name)

**Comment by biosainik** [comment]
Thanks a lot to adam3smith and rmzelle.....

**Comment by rmzelle** [comment]
Could you attach the file http://nitk.ac.in/sites/default/files/public/curriculum/PG-Curriculum2016-revised.pdf to a GitHub comment here? I can't access the website (the site looks to be down: http://downforeveryoneorjustme.com/http://nitk.ac.in/)

**Comment by biosainik** [comment]
@rmzelle. Please find the attachment for the curriculum.
[PG-Curriculum2016-revised.pdf](https://github.com/citation-style-language/styles/files/509072/PG-Curriculum2016-revised.pdf)

**Comment by rmzelle** [comment]
Thanks. I'll take a closer look later and give some more suggestions.

@adam3smith, the reference style format is described at pages 125-126 of the PDF.

File: nitk-thesis.csl [hunk 1]
```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="never">
+  <info>
+    <title>NITK Thesis</title>
```
Comment (rmzelle) [inline nitk-thesis.csl]: We usually don't allow for acronyms in titles. Please change this to `<title>National Institute of Technology Karnataka</title>`

File: nitk-thesis.csl [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    <title-short>NITK</title-short>
+    <id>http://www.zotero.org/styles/nitk-thesis</id>
+    <link href="http://www.zotero.org/styles/nitk-thesis" rel="self"/>
+    <link href="http://nitk.ac.in/sites/default/files/public/curriculum/PG-Curriculum2016-revised.pdf" rel="documentation"/>
```
Comment (rmzelle) [inline nitk-thesis.csl]: Please add a "template" link. See https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link

File: nitk-thesis.csl [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+      <uri>http://arc.nucapt.northwestern.edu/Richard_Karnesky</uri>
+    </contributor>
+    <contributor>
+      <name>Sebastian Karcher</name>
+    </contributor>
+    <contributor>
+      <name>Pragadeesh Sekar</name>
+      <uri>http://www.mendeley.com/profiles/pragadeesh-sekar/</uri>
+      <email>pragadeeshsekar@gmail.com</email>
+    </contributor>
```
Comment (rmzelle) [inline nitk-thesis.csl]: Please remove all contributors except yourself. And either list yourself as a contributor or as an author, but not both. (see also https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)

File: nitk-thesis.csl [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    <category citation-format="author-date"/>
+    <category field="psychology"/>
+    <category field="generic-base"/>
```
Comment (rmzelle) [inline nitk-thesis.csl]: Please change

```
    <category field="psychology"/>
    <category field="generic-base"/>
```

to `<category field="engineering"/>`

File: national-institute-of-technology-karnataka.csl [hunk 5]
```diff
@@ -1,38 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
-<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="never">
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" default-locale="en-GB">
   <info>
-    <title>NITK Thesis</title>
+    <title>National Institute of Technology Karnataka</title>
     <title-short>NITK</title-short>
     <id>http://www.zotero.org/styles/nitk-thesis</id>
```
Comment (rmzelle) [inline national-institute-of-technology-karnataka.csl]: @biosainik, thanks. The style ID also needs to be updated.

Comment (biosainik) [inline national-institute-of-technology-karnataka.csl]: @rmzelle . Thanks for patient and sincere review. Updated the latest correction. Please let me know if I need to change something.

**Comment by rmzelle** [comment]
Thanks. Which CSL style did you start from?

**Comment by biosainik** [comment]
@rmzelle . "American Society for Civil Engineers,. I started with.  My institute style resembles it, more.

**Comment by rmzelle** [comment]
Could you add the line `<link href="http://www.zotero.org/styles/american-society-of-civil-engineers" rel="template"/>`, then?

**Comment by biosainik** [comment]
Done, finally.

**Comment by rmzelle** [comment]
Thanks!

**Comment by biosainik** [comment]
@rmzelle . Thanks a lot. Shall I be able to share this link to my mates now?

**Comment by adam3smith** [comment]
The style will show up in all reference managers that use CSL: In Zotero (on zotero.org/styles ) within an hour, in most others like Mendeley within a day.

**Pull request merged**